### ISSUE TITLE
Unable to use Generics in CPython with Python.NET 

### ISSUE THREAD

**Issue by curzona** [pythonnet/pythonnet#157]
When I use the subscript syntax from http://pythonnet.sourceforge.net/readme.html#generics I get the following error: `TypeError: unsubscriptable object`

With Python 2.7.11 + pythonnet==2.1.0.dev1
```
>python.exe
Python 2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:32:19) [MSC v.1500 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import clr
>>> from System import EventHandler
>>> from System import EventArgs
>>> EventHandler[EventArgs]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsubscriptable object
```

I've also tried pythonnet==2.0.0 and building form github ca15fe8 with ReleaseWin x86 and got the same error.

With IronPython-2.7.5 subscript syntax works:
```
>ipy.exe
IronPython 2.7.5 (2.7.5.0) on .NET 4.0.30319.0 (32-bit)
Type "help", "copyright", "credits" or "license" for more information.
>>> import clr
>>> from System import EventHandler
>>> from System import EventArgs
>>> EventHandler[EventArgs]
<type 'EventHandler[EventArgs]'>
```

**Comment by denfromufa** [comment]
pull requests are welcome :) https://github.com/pythonnet/pythonnet/search?l=csharp&q=unsubscriptable+object&utf8=%E2%9C%93

In both cases I see exceptions, so do you expect to see exactly same message?

**Comment by curzona** [comment]
@denfromufa I've updated my examples.

With IronPython the subscript syntax instantiates the EventHandler as expected but with CPython+Python.NET it returns an error.

**Comment by denfromufa** [comment]
@curzona now I see the problem, this looks like just not implemented in pythonnet, not sure if this workaround would work for your case:

http://stackoverflow.com/questions/8668083/custom-eventhandler-vs-eventhandlereventargs

**Comment by filmor** [comment]
It's definitely implemented, check for example the classes from `System.Collections.Generic`. I guess the problem here is that there are both a generic `EventHandler<ArgsType>` and a non-generic class `EventHandler`, which Python.NET doesn't resolve properly.

You can get the generic class by using its explicit name:

    EventHandler = getattr(System, 'EventHandler`1')
    EventHandler[EventArgs]

**Comment by denfromufa** [comment]
@filmor this is very insightful, pythonnet only imports first instance of collection of overloaded methods/objects, while ironpython imports all of them:

```python
Python 2.7 interactive window [PTVS 2.2.40119.00-14.0]
Type $help for a list of commands.
>>> from System import Action
>>> Action
<types 'Action', 'Action[T]', 'Action[T1, T2]', 'Action[T1, T2, T3]', 'Action[T1, T2, T3, T4]', 'Action[T1, T2, T3, T4, T5]', 'Action[T1, T2, T3, T4, T5, T6]', 'Action[T1, T2, T3, T4, T5, T6, T7]', 'Action[T1, T2, T3, T4, T5, T6, T7, T8]'>
>>> from System import EventHandler
>>> EventHandler
<types 'EventHandler', 'EventHandler[TEventArgs]'>
>>> 
```

**Comment by curzona** [comment]
@filmor Thanks, that works for me.

**Comment by filmor** [comment]
I'll prepare a PR for this, it's actually quite simple to fix.

**Comment by tonyroberts** [comment]
Fixed in PR #164.
